Backend_NodeJS/src/api/controllers/user/auth/register.js

```diff
@@ -0,0 +1,41 @@
+import { User } from "../../../../models/index.js";
+import { validateRegister } from "../../../validators/user.validator.js";
+import bcrypt from "bcrypt";
+
+export default async (req, res) => {
+  try {
+    const { error } = validateRegister(req.body);
+    const { email, password } = req.body;
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    const exists = await User.exists({ email }).catch((err) => {
+      return res.status(500).json({ req, error: err.message });
+    });
+
+    if (exists) return res.status(409).json({ error: "exists" });
+
+    // Hash the password
+    const saltRounds = 10;
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
+
+    let user = new User({
+      email,
+      password: hashedPassword,
+    });
+
+    user = await user.save().catch((err) => {
+      return res.status(500).json({ error: err.message });
+    });
```

Comment:
@bkarakala I could not find firstname and lastname of the user here in registration, is it missing or did i overlook it?